src/GraphQL/Language/AST/AbstractNode.cs

```diff
@@ -17,10 +18,13 @@ public abstract class AbstractNode : INode
         /// </summary>
         public CommentNode CommentNode { get; set; }
 
+        /// <inheritdoc/>
+        public SourceLocation SourceLocation { get; set; }
+
         /// <inheritdoc/>
         public virtual IEnumerable<INode> Children => null;
 
         /// <inheritdoc/>
-        public SourceLocation SourceLocation { get; set; }
+        public virtual void Visit<TState>(Action<INode, TState> action, TState state) { }
```

Comment:
I have specially placed this method everywhere next to `Children`. `Children` is not called for validation at all now but I did not delete it since it is still used by `ComplexityAnalyzer` and `AstPrinter`.